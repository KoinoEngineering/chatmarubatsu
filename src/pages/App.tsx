import MainFrame from "components/templates/MainFrame/MainFrame";
import PageContainer from "components/templates/Page/PageContainer";
import PrivateRoute from "components/templates/PrivateRoute";
import { ConnectedRouter } from "connected-react-router";
import { history } from "core/ConfigureStore";
import FireStoreMonitor from "firebase/FireStoreMonitor";
import React from "react";
import { Redirect, Route as PublicRoute, Switch } from "react-router-dom";
import ROUTES from "utils/routes";
import Lobby from "./Lobby/Lobby";
import Login from "./Login/Login";

const App: React.FC = () => {
    return <MainFrame>
        <PageContainer id="App">
            <ConnectedRouter history={history}>
                <FireStoreMonitor />
                <Switch>
                    <Redirect exact from="/" to={ROUTES.LOGIN} />
                    <PrivateRoute
                        exact
                        path={ROUTES.LOBBY}
                        component={Lobby}
                    />
                    <PublicRoute
                        exact
                        path={ROUTES.LOGIN}
                        component={Login}
                    />
                </Switch>
            </ConnectedRouter>
        </PageContainer>
    </MainFrame>;
};
export default App;