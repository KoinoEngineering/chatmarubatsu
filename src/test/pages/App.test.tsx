import App from "pages/App";
import React from "react";
import { renderWithProvider } from "test/tetutils";

test("snapshot testing", () => {
    const top = renderWithProvider(
        <App />
    );
    expect(top).toMatchSnapshot();
});
