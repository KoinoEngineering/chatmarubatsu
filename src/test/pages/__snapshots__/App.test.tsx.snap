// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="MainFrame"
      >
        <header
          class="MuiPaper-root MuiAppBar-root makeStyles-root-3 MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
          id="Header"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <h1>
                This is Header
              </h1>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <a
                class="makeStyles-anchor-4"
                href="https://github.com/KoinoEngineering/chatmarubatsu"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="makeStyles-icon-5"
                  height="75"
                  role="img"
                  style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
                  viewBox="0 0 16 16"
                  width="75"
                >
                  <path
                    d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z"
                    fill-rule="evenodd"
                  />
                </svg>
              </a>
              <a
                class="makeStyles-anchor-4"
                href="https://www.koino.engineering/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Koino Engineering"
                  class="makeStyles-icon-5"
                  height="75"
                  src="logo.svg"
                />
              </a>
            </div>
          </div>
        </header>
        <div
          class="makeStyles-main-2"
          id="main"
        >
          <div
            class="makeStyles-root-1"
            id="App"
          >
            <div
              class="MuiContainer-root PageContainer-root-150 MuiContainer-maxWidthLg"
              id="Login"
            >
              <div
                class="MuiGrid-root makeStyles-root-148 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
              >
                <div
                  class="MuiPaper-root MuiCard-root makeStyles-root-149 MuiPaper-elevation8 MuiPaper-rounded"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container GridRow-container-160 GridRow-container-162 MuiGrid-justify-xs-center"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1"
                        >
                          ログイン名(任意)を
                        </p>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1"
                        >
                          入力してください
                        </p>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-container GridRow-container-160 GridRow-container-194 MuiGrid-align-items-xs-center"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                          data-shrink="false"
                          for="component-outlined"
                        >
                          Name
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input MuiOutlinedInput-input"
                            id="name"
                            type="text"
                            value=""
                          />
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-252 MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="PrivateNotchedOutline-legendLabelled-254"
                            >
                              <span>
                                Name
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-container GridRow-container-160 GridRow-container-257 MuiGrid-justify-xs-center"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          ログイン
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="MainFrame"
    >
      <header
        class="MuiPaper-root MuiAppBar-root makeStyles-root-3 MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
        id="Header"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <h1>
              This is Header
            </h1>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <a
              class="makeStyles-anchor-4"
              href="https://github.com/KoinoEngineering/chatmarubatsu"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="makeStyles-icon-5"
                height="75"
                role="img"
                style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
                viewBox="0 0 16 16"
                width="75"
              >
                <path
                  d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z"
                  fill-rule="evenodd"
                />
              </svg>
            </a>
            <a
              class="makeStyles-anchor-4"
              href="https://www.koino.engineering/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="Koino Engineering"
                class="makeStyles-icon-5"
                height="75"
                src="logo.svg"
              />
            </a>
          </div>
        </div>
      </header>
      <div
        class="makeStyles-main-2"
        id="main"
      >
        <div
          class="makeStyles-root-1"
          id="App"
        >
          <div
            class="MuiContainer-root PageContainer-root-150 MuiContainer-maxWidthLg"
            id="Login"
          >
            <div
              class="MuiGrid-root makeStyles-root-148 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
            >
              <div
                class="MuiPaper-root MuiCard-root makeStyles-root-149 MuiPaper-elevation8 MuiPaper-rounded"
              >
                <div
                  class="MuiGrid-root MuiGrid-container"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container GridRow-container-160 GridRow-container-162 MuiGrid-justify-xs-center"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1"
                      >
                        ログイン名(任意)を
                      </p>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1"
                      >
                        入力してください
                      </p>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-container GridRow-container-160 GridRow-container-194 MuiGrid-align-items-xs-center"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                        data-shrink="false"
                        for="component-outlined"
                      >
                        Name
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input"
                          id="name"
                          type="text"
                          value=""
                        />
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-252 MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="PrivateNotchedOutline-legendLabelled-254"
                          >
                            <span>
                              Name
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-container GridRow-container-160 GridRow-container-257 MuiGrid-justify-xs-center"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        ログイン
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
